metric_config_parser/config.py

```diff
@@ -264,6 +267,25 @@ def entity_from_path(
         )
 
 
+@attr.s(auto_attribs=True)
+class Repository:
+    """Local repository config files are loaded from."""
+
+    path: Path
+    repo: Repo
+    main_branch: str
+    # indicates whether repository lives in a temporary directory that should be removed on del
+    is_tmp_repo: bool = False
+
+    def __del__(self):
+        # remove the temporary directories repos have been saved in
+        if self.is_tmp_repo:
+            git_dir = Path(self.repo.git_dir)
+            # don't delete repos that come from local directories
+            if git_dir.parent.exists():
+                shutil.rmtree(git_dir.parent)
```

Comment:
to remove temporary repo dirs, but not dirs of repos that come from local directories

Reply:
Does this delete the `tmp_dir` itself or just specific contents within it?

Reply:
It also deletes the directory